keps/sig-apps/2879-ready-pods-job-status/README.md

```diff
@@ -0,0 +1,348 @@
+
+The 99% percentile of Job status updates below 1s, when the controller doesn't
+create new Pods or tracks finishing Pods.
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+- [x] Metrics
+  - Metric name: `job_sync_duration_seconds`, `job_sync_total`.
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+No.
+
+### Dependencies
+
+###### Does this feature depend on any specific services running in the cluster?
+
+No.
+
+### Scalability
+
+###### Will enabling / using this feature result in any new API calls?
+
+
+- API: PUT Job/status
+
+  Estimated throughput: at most one API call for each Job Pod reaching Ready
+  condition.
+  
+  Originating component: job-controller
+
+###### Will enabling / using this feature result in introducing new API types?
+
+No.
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No.
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+- API: Job/status
+
+  Estimated increase in size: New field of less than 10B.
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+No.
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+No.
+
+### Troubleshooting
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+No change from existing behavior of the Job controller.
+
+###### What are other known failure modes?
+
+- When the cluster has apiservers with skewed versions, the `Job.status.ready`
+  might remain zero.
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+1. Check reachability between kube-controller-manager and apiserver.
+1. If the `job_sync_duration_seconds` is too high, check for the number
+   of requests in apiserver coming from the kube-system/job-controller service
+   account. Consider increasing the number of inflight requests for
+   apiserver or tuning [API priority and fairness](https://kubernetes.io/docs/concepts/cluster-administration/flow-control/)
+   to give more priority for the job-controller requests.
+1. If the steps above are insufficient disable the `JobTrackingWithFinalizers`
+   feature gate from apiserver and kube-controller-manager and [report an issue](https://github.com/kubernetes/kubernetes/issues).
+
+## Implementation History
+
+- 2021-08-19: Proposed KEP starting in beta status.
+
+## Drawbacks
+
+The only drawback is an increase in API calls. However, this is capped by
+the number of times a Pod flips ready status. This is usually once for each
+Pod created.
+
+## Alternatives
+
+- Add `Job.status.running`, counting Pods with `Running` phase. The `Running`
```

Comment:
I am wondering the differences between running and ready here.

Reply:
IMO, if the workload is `Ready` detected by readiness probe, it will run into `Running` condition.

Reply:
Correct me if not, I think  it's similar to deployment. `Running` is one phase  of lifecycle, while `ready` means the workload is ready to serve, usually detected by readiness probe.  refer: https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/

Reply:
Will we introduce the ready phase with the help of the readiness probe?

Reply:
I don' think so. As lifecycle is linear, e.g. an old man will never become younger, at least nowadays. but `ready` status will change over time.

Reply:
The `Ready` condition is configurable through the readiness probe. If you don't define any, `Ready` is pretty much equivalent to `Running` phase. I think this is pretty flexible.

Reply:
@alculquicondor perhaps it is worth mentioning that a readiness probe isn't required since it defaults to starting the container process.

Reply:
if readiness probe isn't required, will `Ready` status make `kubers` misunderstood?

Reply:
I don't know what `kubers` is

Reply:
I clarified this paragraph in the PR.

Reply:
😃I mean kubernetes users. Bad descriptions. And I read the updated document again. It's appropriate. Just forget my whispers.

Reply:
> The Ready condition is configurable through the readiness probe. If you don't define any, Ready is pretty much equivalent to Running phase. I think this is pretty flexible.

SGTM, thanks for the explanation.